sdk/core/azure-core/inc/azure/core/internal/http/user_agent.hpp

```diff
@@ -16,6 +16,7 @@ namespace Azure { namespace Core { namespace Http { namespace _detail {
     static std::string GenerateUserAgent(
         std::string const& componentName,
         std::string const& componentVersion,
-        std::string const& applicationId);
+        std::string const& applicationId,
+        long cplusplusValue);
```

Comment:
Why long? Also, should this be a string instead?

Reply:
I strongly suspect that the reason this is a long is that the language defines __cplusplus as a numeric constant and specifies numeric comparisons for it.

But Anton could better answer that.

Reply:
Larry is right - `__cplusplus` macro is defined as `long` constant.